File: src/server/src/main/java/com/project/application/models/exercise/ExerciseResource.java
package com.project.application.models.exercise;

import com.project.domain.workout.Exercise;
import io.micronaut.serde.annotation.Serdeable;
import jakarta.annotation.Nullable;

import java.util.UUID;

@Serdeable
public record ExerciseResource(UUID id, @Nullable String imageUrl, @Nullable String name, @Nullable String translatedName) {
    public static ExerciseResource from(Exercise exercise) {
        return new ExerciseResource(exercise.getId(), exercise.getImageUrl(), exercise.getName(), exercise.getTranslatedName());
    }
}

File: src/server/src/main/kotlin/com/project/application/models/exercise/ExerciseResource.kt
package com.project.application.models.exercise

import com.project.domain.workout.Exercise
import io.micronaut.serde.annotation.Serdeable
import java.util.UUID

@Serdeable
data class ExerciseResource(val id: UUID, val imageUrl: String?, val name: String?, val translatedName: String?) {
    companion object {
        fun from(exercise: Exercise): ExerciseResource {
            return ExerciseResource(exercise.id, exercise.imageUrl, exercise.name, exercise.translatedName)
        }
    }
}
